src/moscot/utils/tagged_array.py

```diff
@@ -159,3 +172,205 @@ def is_kernel(self) -> bool:
+        adata: AnnData,
+        a: ArrayLike,
+        b: ArrayLike,
+        xy_attr: Literal["X", "obsp", "obsm", "layers", "uns"] = None,
+        xy_key: Optional[str] = None,
+        xy_cost: CostFn_t = "sq_euclidean",
+        xx_attr: Literal["X", "obsp", "obsm", "layers", "uns"] = None,
+        xx_key: Optional[str] = None,
+        xx_cost: CostFn_t = "sq_euclidean",
+        conditions_attr: Optional[Literal["obs", "var", "obsm", "varm", "layers", "uns"]] = None,
+        conditions_key: Optional[str] = None,
+        backend: Literal["ott"] = "ott",
+        **kwargs: Any,
+    ) -> "DistributionContainer":
+        """Create distribution container from :class:`~anndata.AnnData`.
+
+        .. warning::
+            Sparse arrays will be always densified.
+
+        Parameters
+        ----------
+        adata
+            Annotated data object.
+        a
+            Marginals when used as source distribution.
+        b
+            Marginals when used as target distribution.
+        xy_attr
+            Attribute of :paramref:`adata` containing the data for the shared space.
+        xy_key
+            Key of :paramref:`xy_attr` containing the data for the shared space.
+        xy_cost
+            Cost function when in the shared space.
+        xx_attr
+            Attribute of :paramref:`adata` containing the data for the incomparable space.
+        xx_key
+            Key of :paramref:`xx_attr` containing the data for the incomparable space.
+        xx_cost
+            Cost function in the incomparable space.
+        conditions_attr
+            Attribute of :paramref:`adata` containing the conditions.
+        conditions_key
+            Key of :paramref:`conditions_attr` containing the conditions.
+        backend
+            Backend to use.
+        kwargs
+            Keyword arguments to pass to the cost functions.
+
+        Returns
+        -------
+        The distribution container.
+        """
+        contains_linear, contains_quadratic, contains_condition = cls._verify_input(
+            xy_attr, xy_key, xx_attr, xx_key, conditions_attr, conditions_key
+        )
+
+        if contains_linear:
+            xy_data = cls._extract_data(adata, attr=xy_attr, key=xy_key)
+            xy_cost_fn = get_cost(xy_cost, backend=backend, **kwargs)
+        else:
+            xy_data = None
+            xy_cost_fn = None
+
+        if contains_quadratic:
+            xx_data = cls._extract_data(adata, attr=xx_attr, key=xx_key)
+            xx_cost_fn = get_cost(xx_cost, backend=backend, **kwargs)
+        else:
+            xx_data = None
+            xx_cost_fn = None
+
+        conditions_data = (
+            cls._extract_data(adata, attr=conditions_attr, key=conditions_key) if contains_condition else None  # type: ignore[arg-type]  # noqa:E501
+        )
+        return cls(xy=xy_data, xx=xx_data, a=a, b=b, conditions=conditions_data, cost_xy=xy_cost_fn, cost_xx=xx_cost_fn)
+
+
+class DistributionCollection:
```

Comment:
I wonder how useful this class is, since it's just a like read-only dict with custom repr.
If you're planning to add more functionality to it (maybe like some sanity checks), would keep it.